controller/src/main/java/io/pravega/controller/eventProcessor/impl/ConcurrentEventProcessor.java

```diff
@@ -46,7 +46,7 @@
 @Slf4j
 public class ConcurrentEventProcessor<R extends ControllerEvent, H extends RequestHandler<R>>
         extends EventProcessor<R> {
-    private static final int MAX_CONCURRENT = 10000;
+    private static final int MAX_CONCURRENT = 1000;
```

Comment:
by reducing concurrent processing of events by 10x, in peak load conditions, the memory footprint also goes down by 10x. 
10000 is a lot anyway. If each processing has a footprint of only 100kb then with 10k events being picked and pocessed, the load on memory will be 1Gb.
the actual footprint will be higher because events for which processing has completed will have their footprint shaved off only after a GC cycle. 

It will impact our throughput of events processed. but these are background asynchronous tasks anyway and we are processing 1000 events concurrently as opposed to 10k. 
We can have reasonably large number of segments on our internal streams and then the load gets be distributed across multiple controller instances to achieve high throughput. 
but we dont want to overwhelm a single controller instance by picking more work than its memory can reasonably sustain. hence 1000 seemed like a reasonable number even for an average 1mb memory footprint for each event's processing.